shm/src/ship.cpp

```diff
@@ -42,3 +43,42 @@ size_t Ship::getAvailableSpace() const {
                                            });
     return capacity_ - reservedSpace;
 }
+
+void Ship::load(const std::shared_ptr<Cargo>& cargo) {
+    if (cargo->getAmount() > getAvailableSpace()) {
+        std::cerr << "not enough space on the ship";
+        return;
+    }
+    auto cargoOnShip = std::find_if(cargo_.begin(), cargo_.end(),
+                                    [name = cargo ->getName()](const auto& el) {
+                                        return el->getName() == name;
+                                    });
+    if (cargoOnShip == cargo_.end()) {
+        cargo_.push_back(cargo);
+    } else {
+        (*cargoOnShip)->operator+=(cargo->getAmount());
+    }
+}
+
+void Ship::unload(const Cargo* const& cargo) {
+    auto cargoOnShip = std::find_if(cargo_.begin(), cargo_.end(),
+                                    [name = cargo ->getName()](const auto& el) {
+                                        return el->getName() == name;
+                                    });
+    if (cargoOnShip == cargo_.end()) {
+        std::cerr << "Not found cargo";
+        return;
+    }
+    if ((*cargoOnShip)->getAmount() < cargo->getAmount()) {
+        std::cerr << "You don't have enough cargo";

```

Comment:
new line
```suggestion
        std::cerr << "You don't have enough cargo";
```suggestion
        std::cerr << "You don't have enough cargo\n";

```
```